src/windows-emulator/minidump_loader.cpp

```diff
@@ -0,0 +1,758 @@
+        for (const auto& module : modules)
+        {
+            win_emu.log.info("Module: %s at 0x%" PRIx64 " (%u bytes)\n", module.module_name.c_str(),
+                             module.base_of_image, module.size_of_image);
+        }
+
+        // Process threads
+        const auto& threads = dump_file->threads();
+        for (const auto& thread : threads)
+        {
+            win_emu.log.info("Thread %u: TEB 0x%" PRIx64 ", stack 0x%" PRIx64 " (%u bytes), context %u bytes\n",
+                             thread.thread_id, thread.teb, thread.stack_start_of_memory_range, thread.stack_data_size,
+                             thread.context_data_size);
+        }
+
+        // Process handles
+        const auto& handles = dump_file->handles();
+        if (!handles.empty())
+        {
+            std::map<std::string, size_t> handle_type_counts;
+            for (const auto& handle : handles)
+            {
+                handle_type_counts[handle.type_name]++;
+            }
+            win_emu.log.info("Handles: %zu total\n", handles.size());
+            for (const auto& [type, count] : handle_type_counts)
+            {
+                win_emu.log.info("  %s: %zu\n", type.c_str(), count);
+            }
+        }
+
+        // Process exception info
+        const auto* exception = dump_file->get_exception_info();
+        if (exception)
+        {
+            win_emu.log.info("Exception: thread %u, code 0x%08X at 0x%" PRIx64 "\n", exception->thread_id,
+                             exception->exception_record.exception_code, exception->exception_record.exception_address);
+        }
+    }
+
+    void reconstruct_memory_state(windows_emulator& win_emu, const minidump::minidump_file* dump_file,
+                                  minidump::minidump_reader* dump_reader)
+    {
+        if (!dump_file || !dump_reader)
+        {
+            win_emu.log.error("Dump file or reader not loaded\n");
+            return;
+        }
+
+        const auto& memory_regions = dump_file->memory_regions();
+        const auto& memory_segments = dump_file->memory_segments();
+
+        win_emu.log.info("Reconstructing memory: %zu regions, %zu data segments\n", memory_regions.size(),
+                         memory_segments.size());
+        size_t reserved_count = 0;
+        size_t committed_count = 0;
+        size_t failed_count = 0;
+
+        for (const auto& region : memory_regions)
+        {
+            const bool is_reserved = (region.state & MEM_RESERVE) != 0;
+            const bool is_committed = (region.state & MEM_COMMIT) != 0;
+            const bool is_free = (region.state & MEM_FREE) != 0;
+
+            if (is_free)
+            {
+                continue;
+            }
+
+            memory_permission perms = memory_permission::none;
+
+            // Strip modifier flags like PAGE_GUARD, PAGE_NOCACHE, etc.
+            const uint32_t base_protect = region.protect & 0xFF;
```

Comment:
why not reuse this https://github.com/momo5502/sogen/blob/main/src/windows-emulator/memory_utils.hpp#L22 ?